design/ceph/object/bucketpolicy.md

```diff
@@ -0,0 +1,93 @@
+---
+title: adding bucket policy for ceph object store
+target-version: release-1.4
+---
+
+# Feature Name
+Adding bucket policy support for ceph object store
+
+## Summary
+The bucket policy is the feature in which permissions for specific user can be set on s3 bucket. Read more about it from [ceph documentation](https://docs.ceph.com/docs/master/radosgw/bucketpolicy/)
+
+Currently [ceph object store](/Documentation/ceph-object.md) can be consumed either via [OBC](/Documentation/ceph-object-bucket-claim.md) and [ceph object user](/Documentation/ceph-object-store-user-crd.md). As of now there is no direct way for ceph object user to access the OBC. The idea behind this feature to allow that functionality via Rook. Refer bucket policy examples from [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/example-bucket-policies.html).  Please note it is different from `IAM policies`.
+
+
+## Proposal details
+
+Followings CRDs are needed to add for defining policies:
+  - _bucketPolicy_ in the `Spec` section of `CephObjectStoreUser`
+  - _bucketPolicy_ in the `parameters` section of `StorageClass` for `ObjectBucketClaim`
+
+Policies need to provide in generic `json` format. A policy can have multiple `statement`.
+
+The following checks are needed to perform in Rook to verify whether the `bucketpolicy` applicable to OBC.
+- for ceph object user, `Principal` value should have username and `Resource` should have specific bucket names. It can be defined for buckets which are not part of the OBC as well, the `bucketname` of an OBC can be fetched from its `configmap`.
+- In `StorageClass`, `Principal` value should be `*`(applicable to all users) and `Resource` should have the bucket name can be empty since it can be generated name from Rook as well and will be attached before setting the policy.
+
+Examples:
+```yaml
+# storageclass-bucket.yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: rook-ceph-delete-bucket
+provisioner: rook-ceph.ceph.rook.io/bucket
+# set the reclaim policy to delete the bucket and all objects
+# when its OBC is deleted.
+reclaimPolicy: Delete
+parameters:
+  objectStoreName: my-store
+  objectStoreNamespace: rook-ceph
+  region: us-east-1
+  bucketName: ceph-bkt
+  bucketPolicy: "Version": "2012-10-17",
+		"Statement": [{
+		"Effect": "Allow",
```

Comment:
nit: can you use spaces instead of tabs to avoid this rendering?

Reply:
will address in next PR